commons/src/test/java/io/aiven/kafka/tiered/storage/commons/chunkindex/ChunkTest.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright 2023 Aiven Oy
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.aiven.kafka.tiered.storage.commons.chunkindex;
+
+import org.junit.jupiter.api.Test;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class ChunkTest {
+    @Test
+    void identical() {
+        final var c1 = new Chunk(0, 20, 30, 100, 120);
+        final var c2 = new Chunk(0, 20, 30, 100, 120);
+        assertThat(c1.equals(c2)).isTrue();
+        assertThat(c2.equals(c1)).isTrue();
+        assertThat(c1.hashCode()).isEqualTo(c2.hashCode());
```

Comment:
```suggestion
        assertThat(c1).hasSameHashCodeAs(c2);
```
And there is also `doesNotHaveSameHashCodeAs` for other places